### PR TITLE
Update CHANGELOG.md for 2.3.0

### PR DESCRIPTION
**Pull request by jforissier**
Update change log, preparing for release 2.3.0.

Signed-off-by: Jerome Forissier <jerome.forissier@linaro.org>

### CODE REVIEW

**Comment by jforissier** [comment]
First try at the release notes for the upcoming 2.3.0 release of OP-TEE.
As usual, I'll merge this just before the release, after the test reports are in and any issues are possibly fixed. Feel free to comment any time.

**Comment by jforissier** [comment]
@MrVan did you run xtest with the GP tests or without?

**Comment by sorenb-xlnx** [comment]
For zynqmp using default xtest:
`Tested-by: Sören Brinkmann <soren.brinkmann@xilinx.com>`

**Comment by jbech-linaro** [comment]
MTK8173 with all latest from this morning:
```bash
root@MT8173_EVB:/ xtest -l 15
...
46492 subtests of which 0 failed
70 test cases of which 0 failed
0 test case was skipped
TEE test application done!
```
So, the MTK8173 device is back on track.
`Tested-by: Joakim Bech <joakim.bech@linaro.org> (MTK8173)`

(I'm doing one more test build, to see if we can get rid of the `arm64: dt: Remove timer from mt8173 **not for mainline**` which I added a long time ago).

**Comment by jenswi-linaro** [comment]
Test both with our present https://github.com/linaro-swg/linux/tree/optee brach and also with https://github.com/jenswi-linaro/linux-1/tree/optee-next

`Tested-by: Jens Wiklander <jens.wiklander@linaro.org> (Juno)`

**Comment by jforissier** [comment]
For HiKey using "advanced" xtest (with GP tests):
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey)`

Tested with https://github.com/linaro-swg/linux/tree/optee in various 32/64-bit combinations + pager. Also tested one config with https://github.com/jenswi-linaro/linux-1/tree/optee-next (based on 4.9). The 4.9 kernel has a slightly annoying issue which does not seem related to OP-TEE:
```
[  242.819370] INFO: task kworker/4:0:36 blocked for more than 120 seconds.
[  242.826104]       Not tainted 4.9.0-00152-g1bdaf42 #10
[  242.831250] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[  242.839146] kworker/4:0     D    0    36      2 0x00000000
[  242.844657] Workqueue: events_freezable mmc_rescan
[  242.849458] Call trace:
[  242.851916] [<ffff000008085550>] __switch_to+0x90/0xa4
[  242.857063] [<ffff0000088e1650>] __schedule+0x178/0x564
[  242.862294] [<ffff0000088e1a78>] schedule+0x3c/0xa0
[  242.867181] [<ffff0000088e47cc>] schedule_timeout+0x148/0x218
[  242.872934] [<ffff0000088e25b8>] wait_for_common+0xb4/0x158
[  242.878512] [<ffff0000088e2670>] wait_for_completion+0x14/0x1c
[  242.884348] [<ffff00000875e540>] mmc_wait_for_req_done+0x38/0x11c
[  242.890446] [<ffff00000875e6e0>] mmc_wait_for_req+0xbc/0xf8
[  242.896023] [<ffff00000875e78c>] mmc_wait_for_cmd+0x70/0x88
[  242.901604] [<ffff000008769550>] mmc_io_rw_direct_host+0x78/0x100
[  242.907702] [<ffff000008769970>] sdio_reset+0x28/0x70
[  242.912757] [<ffff00000876105c>] mmc_rescan+0x2e8/0x34c
[  242.917990] [<ffff0000080d3258>] process_one_work+0x128/0x2e4
[  242.923743] [<ffff0000080d346c>] worker_thread+0x58/0x434
[  242.929148] [<ffff0000080d8f24>] kthread+0xd4/0xe8
[  242.933945] [<ffff000008082e80>] ret_from_fork+0x10/0x50
```

**Comment by igoropaniuk** [comment]
RPi3 (using `rpi3.xml` manifest from scratch):

```
root@RPi3:/bin ./xtest -l 15
...
46493 subtests of which 0 failed
70 test cases of which 0 failed
0 test case was skipped
TEE test application done!
```

Also tested TFTP/NFS boot (by following updated steps in https://github.com/OP-TEE/build/blob/master/docs/rpi3.md)
`Tested-by: Igor Opaniuk <igor.opaniuk@linaro.org> (RPi3)`

**Comment by vchong** [comment]
HiKey kernel issue above is here: https://bugs.96boards.org/show_bug.cgi?id=467

**Comment by jbech-linaro** [comment]
HiKey Debian:
`Tested-by: Joakim Bech <joakim.bech@linaro.org> (HiKey Debian)`

With latest 4.9 kernel on linaro-swg/linux/optee and recent patch, just merged in build.git (https://github.com/OP-TEE/build/pull/100)
```bash
+-----------------------------------------------------
23537 subtests of which 0 failed
70 test cases of which 0 failed
0 test case was skipped
TEE test application done!
```

**Comment by jenswi-linaro** [comment]
`Tested-by: Jens Wiklander <jens.wiklander@linaro.org> (FVP)`

**Comment by jforissier** [comment]
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (D02)`

**Comment by lorc** [comment]
`Tested-by: Volodymyr Babchuk <vlad.babchuk@gmail.com> (RCAR H3)`

**Comment by b49020** [comment]
For LS1021A using standard tests:
`Tested-by: Sumit Garg <sumit.garg@nxp.com>`

`+-----------------------------------------------------
23537 subtests of which 0 failed
70 test cases of which 0 failed
0 test case was skipped
TEE test application done!`

**Comment by jbech-linaro** [comment]
Texas Instruments DRA7xx, PLATFORM=ti-dra7xx (@glneo FYI)
`Tested-by: Joakim Bech <joakim.bech@linaro.org> (dra7)`
```bash
root@DRA7xx:/ xtest -l 15
...
+-----------------------------------------------------
46474 subtests of which 0 failed
69 test cases of which 0 failed
0 test case was skipped
TEE test application done!
root@DRA7xx:/ 
```

**Comment by etienne-lms** [comment]
`Tested-by: Etienne Carriere <etienne.carriere@linaro.org> (b2120/b2260/qemu, with GP)`

**Comment by jforissier** [comment]
Tested OK on QEMU v8 (64-bit).

`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (vexpress-qemu_armv8a)`

Unfortunately the current tip of the QEMU master branch appears to be broken (ARM-TF logs: "VERBOSE: Skipping CPU specific reset operation for non-matching CPU revision number."). So I think we'll have to pin v2.8.0 instead.

**Comment by jforissier** [comment]
The QEMU Aarch64 regression has already been reported and the root cause identified, see: http://lists.nongnu.org/archive/html/qemu-devel/2017-01/msg02704.html. A fix should appear shortly.

**Comment by jforissier** [comment]
Thanks everyone for testing.
Commits squashed and rebased on top of master, ready for merging tomorrow.
BTW, the QEMU regression was fixed: the upstream master branch works fine.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,131 @@
...
+  the format defined in RFC4122 (a missing hyphen was added). The old format
+  is still supported, but deprecated, and will likely be removed with the
+  next major release.
+
+* Drop write permission to non-writable ELF segments after TA loading is
+  complete.
+
+* mm: fix confusing memory mapping debug traces
+
+* plat-ti: fix issues fiwn MMU mapping
```
Comment (jenswi-linaro) [inline CHANGELOG.md]: fiwn?

Comment (jforissier) [inline CHANGELOG.md]: Oops, will fix. Thanks.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,131 @@
...
+
+* Add debug assertions to spinlocks and mutexes
+
+* Add more CP15 register access macros for Cortex-A9
+
+* armv7: redesign secure monitor to make it easier to register services
+
+* armv7: cleanup boot arguments
+
+* libteec: extend `TEE_CheckMemoryAccessRights()` with
```
Comment (etienne-lms) [inline CHANGELOG.md]: typo: this is `libutee`, not `libteec`

Comment (jforissier) [inline CHANGELOG.md]: So true! Will fix, thanks.

**Comment by jforissier** [comment]
Updated.

**Comment by jbech-linaro** [comment]
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

**Pull request merged**